nasa/earthdata-search
# EDSC-4050: Added ability to generate L2 Raster Products using SWODLR in EDSC.

# Overview

### What is the feature?

Have an endpoint/lambda to process SWODLR orders in earthdata search.

### What is the Solution?

Added the endpoint/lambda to process SWODLR orders in earthdata search.

### What areas of the application does this impact?

List impacted areas.

# Testing

### Reproduction steps

- **Environment for testing:** 
- **Collection to test with:** C2799438271-POCLOUD

1. go to localhost:8080/search/granules?ee=prod&p=C2799438271-POCLOUD
2. Add the 2 granules  
  1. G2938391118-POCLOUD (SWOT_L2_HR_Raster_250m_UTM34L_N_x_x_x_013_514_087F_20240414T225546_20240414T225558_PIC0_01)
  2. G2938390924-POCLOUD (SWOT_L2_HR_Raster_250m_UTM34L_N_x_x_x_013_514_086F_20240414T225526_20240414T225547_PIC0_01)
4. Update the db entry in *retrieval_orders* and change the values to match
`granule_params: {
    exclude: {},
    options: {},
    page_num: 1,
    sort_key: '-start_date',
    json_data: {
      params: {
        rasterResolution: 6,
        outputSamplingGridType: 'GEO',
        outputGranuleExtentFlag: false
      },
      custom_params: {
        'G2938391118-POCLOUD': {
          utmZoneAdjust: null,
          mgrsBandAdjust: null
        },
        'G2938390924-POCLOUD': {
          utmZoneAdjust: null,
          mgrsBandAdjust: null
        }
      }
    },
    page_size: 2000,
    concept_id: [
      'G2938390910-POCLOUD',
      'G2938390924-POCLOUD'
    ],
    echo_collection_id: 'C2799438271-POCLOUD',
    two_d_coordinate_system: {}
  }`

5. You will also need to edit the url of the service used in the `retrieval_collections` table under the `access_method` column

6. check https://github.com/nasa/earthdata-search/pull/1739/files#diff-93cc28b3641d217b6bb7b578b51ca669f0861b5aa021f358de158e14b11da700 and save it with the id equal to the *retrieval_orders* entry id.

7. run `npm run invoke-local -- --function submitSwodlrOrder -p /Users/bporeh/gsfc/earthdata-search/payloads/mock_swodlr_lambda_1.json` switching the mock_swodlr_lambda_1.json to the file referenced in step 

8. ensure the lambda runs all the way.

9. Subsequent requests to get the status can be sent to the swoldr API along with the productId that it generates in the response

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist
